src/pages/billing/CreateAzureBillingProjectModal.js

```diff
@@ -0,0 +1,151 @@
+            setChosenManagedApp('')
+            setErrorFetchingManagedApps(true)
+            // We can't rely on the formatting of the error, so show a generic message but include the error in the console for debugging purposes.
+            const error = await (obj instanceof Response ? obj.text() : obj)
+            console.error(error)
+          }
+        }
+      )
+      fetchManagedApps()
+    }
+  }, [isValidSubscriptionId, subscriptionId, signal])
+
+  const subscriptionIdError = Utils.cond(
+    [subscriptionIdTouched && !isValidSubscriptionId, () => Utils.summarizeErrors(subscriptionIdErrors?.subscriptionId)],
+    [errorFetchingManagedApps, () => `Unable to retrieve Managed Applications for that subscription`],
+    [errorFetchingManagedApps === false && managedApps.length === 0, () => `No Terra Managed Applications exist for that subscription`],
+    [Utils.DEFAULT, () => undefined]
+  )
+
+  return h(Modal, {
+    onDismiss,
+    shouldCloseOnOverlayClick: false,
+    title: 'Create Terra Billing Project',
+    okButton:
+      h(ButtonPrimary, {
+        disabled: billingProjectNameErrors || !isValidSubscriptionId || !chosenManagedApp,
+        onClick: submit
+      }, ['Create'])
+  }, [
+    h(Fragment, [
+      h(IdContainer, [id => h(Fragment, [
+        h(FormLabel, { htmlFor: id, required: true }, ['Terra billing project']),
+        h(ValidatedInput, {
+          inputProps: {
+            id,
+            autoFocus: true,
+            value: billingProjectName,
+            placeholder: 'Enter a name',
+            onChange: v => {
+              setBillingProjectName(v)
+              setBillingProjectNameTouched(true)
+            }
+          },
+          error: billingProjectNameTouched && Utils.summarizeErrors(billingProjectNameErrors?.billingProjectName)
+        })
+      ])]),
+      !(billingProjectNameTouched && billingProjectNameErrors) && formHint('Name must be unique and cannot be changed.'),
+      h(IdContainer, [id => h(Fragment, [
+        h(FormLabel, { htmlFor: id, required: true }, ['Azure subscription']),
+        h(ValidatedInput, {
+          inputProps: {
+            id,
+            autoFocus: true,
+            placeholder: 'Enter an Azure Subscription ID',
+            onChange: v => {
+              setErrorFetchingManagedApps(undefined)
+              setSubscriptionId(v)
+              setSubscriptionIdTouched(true)
+            }
+          },
+          error: subscriptionIdError
+        })
+      ])]),
+      h(IdContainer, [id => h(Fragment, [
+        h(FormLabel, { htmlFor: id, required: true }, ['Managed application']),
+        div({ style: { fontSize: 14 } }, [
+          h(Select, {
+            id,
+            isMulti: false,
+            placeholder: 'Select a managed application',
+            isDisabled: errorFetchingManagedApps !== false || !!subscriptionIdError,
+            value: chosenManagedApp,
+            onChange: ({ value }) => setChosenManagedApp(value),
```

Comment:
Not a suggestion, just something I tried to see how it would look and sharing the result.
```javascript
            onChange: _.flow(_.get('value'), setChosenManagedApp),
```

Reply:
To me this is less clear/more difficult to read.